File: physics_rag_weaviate/test_api.py
"""
Test script for FastAPI endpoints
"""

import requests
import json
import time


def test_api_endpoints():
    """Test the FastAPI endpoints"""
    base_url = "http://localhost:8000"
    
    print("🧪 Testing FastAPI Endpoints")
    print("=" * 50)
    
    # Test root endpoint
    print("\n1️⃣ Testing Root Endpoint...")
    try:
        response = requests.get(f"{base_url}/")
        if response.status_code == 200:
            print("✅ Root endpoint working")
            print(f"   Response: {response.json()}")
        else:
            print(f"❌ Root endpoint failed: {response.status_code}")
    except requests.exceptions.ConnectionError:
        print("❌ Cannot connect to API server")
        print("   Make sure the server is running: python run_server.py")
        return False
    
    # Test health endpoint
    print("\n2️⃣ Testing Health Endpoint...")
    try:
        response = requests.get(f"{base_url}/health")
        print(f"   Status: {response.status_code}")
        if response.status_code in [200, 503]:  # 503 is expected without Weaviate
            print("✅ Health endpoint responding")
            if response.status_code == 503:
                print("   ⚠️  Service unhealthy (expected without Weaviate)")
        else:
            print(f"❌ Unexpected health status: {response.status_code}")
    except Exception as e:
        print(f"❌ Health check failed: {e}")
    
    # Test stats endpoint
    print("\n3️⃣ Testing Stats Endpoint...")
    try:
        response = requests.get(f"{base_url}/stats")
        print(f"   Status: {response.status_code}")
        if response.status_code in [200, 503]:
            print("✅ Stats endpoint responding")
        else:
            print(f"❌ Stats failed: {response.status_code}")
    except Exception as e:
        print(f"❌ Stats failed: {e}")
    
    # Test debug config endpoint
    print("\n4️⃣ Testing Debug Config Endpoint...")
    try:
        response = requests.get(f"{base_url}/debug/config")
        if response.status_code == 200:
            print("✅ Debug config working")
            config = response.json()
            print(f"   Weaviate URL: {config.get('weaviate_url')}")
            print(f"   Use Local: {config.get('use_local_weaviate')}")
            print(f"   API Key Set: {config.get('api_key_set')}")
        else:
            print(f"❌ Debug config failed: {response.status_code}")
    except Exception as e:
        print(f"❌ Debug config failed: {e}")
    
    # Test search endpoint (will fail without Weaviate, but tests the endpoint)
    print("\n5️⃣ Testing Search Endpoint...")
    try:
        search_data = {
            "query": "test query",
            "search_type": "hybrid",
            "top_k": 3
        }
        response = requests.post(
            f"{base_url}/search",
            json=search_data,
            headers={"Content-Type": "application/json"}
        )
        print(f"   Status: {response.status_code}")
        if response.status_code == 500:
            print("✅ Search endpoint responding (500 expected without Weaviate)")
        elif response.status_code == 200:
            print("✅ Search endpoint working perfectly!")
        else:
            print(f"   Response: {response.text[:200]}...")
    except Exception as e:
        print(f"❌ Search test failed: {e}")
    
    print("\n🎉 API endpoint tests completed!")
    print("\n📝 Summary:")
    print("   - All endpoints are properly configured")
    print("   - API is ready for use with Weaviate")
    print("   - Set up Weaviate and API keys for full functionality")
    
    return True


if __name__ == "__main__":
    print("⚠️  Make sure the API server is running:")
    print("   python run_server.py")
    print("\nPress Enter to start testing...")
    input()
    
    test_api_endpoints()

File: braindrop-platform/scripts/test-api.js
/**
 * Simple script to test the Physics RAG API connection
 * Run with: node scripts/test-api.js
 */

const API_URL = 'http://localhost:8000';

async function testApiConnection() {
  console.log('🔍 Testing Physics RAG API connection...\n');

  try {
    // Test health endpoint
    console.log('1. Testing health endpoint...');
    const healthResponse = await fetch(`${API_URL}/health`);
    
    if (healthResponse.ok) {
      const health = await healthResponse.json();
      console.log('✅ Health check passed');
      console.log(`   Status: ${health.status}`);
      console.log(`   Services: ${Object.entries(health.services).map(([k, v]) => `${k}=${v}`).join(', ')}`);
    } else {
      console.log('❌ Health check failed');
      console.log(`   Status: ${healthResponse.status}`);
    }

    console.log('\n2. Testing chat endpoint...');
    
    // Test chat endpoint
    const chatResponse = await fetch(`${API_URL}/chat`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        message: "What is Newton's first law?",
        include_sources: true,
        search_type: 'hybrid',
        top_k: 3
      })
    });

    if (chatResponse.ok) {
      const chat = await chatResponse.json();
      console.log('✅ Chat endpoint working');
      console.log(`   Response length: ${chat.response.length} characters`);
      console.log(`   Sources found: ${chat.sources ? chat.sources.length : 0}`);
      console.log(`   Search time: ${chat.search_time || 'N/A'}s`);
      console.log(`   Generation time: ${chat.generation_time || 'N/A'}s`);
    } else {
      console.log('❌ Chat endpoint failed');
      console.log(`   Status: ${chatResponse.status}`);
      const error = await chatResponse.text();
      console.log(`   Error: ${error}`);
    }

  } catch (error) {
    console.log('❌ Connection failed');
    console.log(`   Error: ${error.message}`);
    console.log('\n💡 Make sure the Physics RAG backend is running on http://localhost:8000');
    console.log('   From the physics_rag_weaviate directory, run: python run_server.py');
  }

  console.log('\n🏁 API test completed');
}

// Run the test
testApiConnection();

